experimental/oak_baremetal_kernel/src/elf.rs

```diff
@@ -0,0 +1,33 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use goblin::elf64::program_header::ProgramHeader;
+use x86_64::VirtAddr;
+
+/// Interpret raw memory at the given address as an ELF header and return the program headers.
+///
+/// Safety: this virtual address must be valid and contain ELF headers and program headers.
```

Comment:
The "Safety" reads more like a pre-condition on the virtual address than an argument why this is safe to do. Or do I misunderstand?

Reply:
No, that's exactly what it is -- a precondition. The function is marked unsafe because we can't check that, so it's up to the caller to guarantee that.